src/ng/q.js

```diff
@@ -504,6 +504,21 @@ function qFactory(nextTick, exceptionHandler) {
    * Combines multiple promises into a single promise that is resolved when all of the input
    * promises are resolved.
    *
+   * ```js
+   *   promiseC = $q.all([promiseA, promiseB]).then(function(result) {
+   *     // success: do something and resolve promiseC
+   *     return result;
+   *   }, function(reason) {
+   *     // error: handle the error if possible and
+   *     //        resolve promiseC with newPromiseOrValue,
+   *     //        otherwise forward the rejection to promiseC
+   *     if (canHandle(reason)) {
+   *      // handle the error and recover
+   *      return newPromiseOrValue;
+   *     }
+   *   });
```

Comment:
I think you should also show how to properly handle the case where `canHandle()` returns false (i.e. propagate the rejection).